src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs

```diff
@@ -231,15 +231,11 @@ public Task WaitForMeasurementsAsync(int minCount, CancellationToken cancellatio
             {
                 lock (_measurements)
                 {
-                    _waiters.Remove(w);
+                    _ = _waiters.Remove(w);
                 }
 
                 // trigger the task from outside the lock
-#if NET6_0_OR_GREATER
-                w.TaskSource.TrySetCanceled(cancellationToken);
-#else
-                w.TaskSource.TrySetCanceled();
-#endif
+                _ = w.TaskSource.TrySetCanceled(cancellationToken);
```

Comment:
Shouldn't we await this? This is testing code, and discarding a task will ignore potential exceptions in TrySetCanceled.

Reply:
Wait for what? TrySetCanceled returns a boolean. And this is intentionally fire-and-forget so we don't care about the return value.